data_structures/binaryheap.cpp

```diff
@@ -13,10 +13,14 @@ class MinHeap {
     int heap_size;  ///< Current number of elements in min heap
 
  public:
-    /** Constructor
+    /** Constructor: Builds a heap from a given array a[] of given size
      * \param[in] capacity initial heap capacity
      */
-    MinHeap(int capacity);
+    explicit MinHeap(int cap) {
+        heap_size = 0;
+        capacity = cap;
+        harr = new int[cap];
+    }
 
     /** to heapify a subtree with the root at given index
      */
```

Comment:
please delete this line

Reply:
removed

Reply:
> removed

Oh, my bad. I misread the function name and thought it was the same as the constructor. The commit 48a163e51ab53a73c9edbb02633e6a3673865cea is the last good commit. Please revert to it. Thank you

Reply:
no problem, i too did not looked closely, i thought you were right